src/v/datalake/arrow_translator.cc

```diff
@@ -0,0 +1,805 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "datalake/arrow_translator.h"
+
+#include "container/fragmented_vector.h"
+#include "iceberg/datatypes.h"
+
+#include <arrow/api.h>
+#include <arrow/array/array_primitive.h>
+#include <arrow/array/builder_base.h>
+#include <arrow/array/builder_binary.h>
+#include <arrow/array/builder_decimal.h>
+#include <arrow/array/builder_nested.h>
+#include <arrow/array/builder_primitive.h>
+#include <arrow/type.h>
+#include <arrow/type_fwd.h>
+#include <arrow/type_traits.h>
+
+#include <memory>
+#include <stdexcept>
+
+namespace datalake {
+class schema_converter_interface {
+public:
```

Comment:
was there a file rename? if so, break the rename out as a separate commit please

Reply:
Done